[Justin Yeo Jun Chen] iP

Justin-Yeo opened the pull request:
# Justbot :trollface: : 

> "I am a cool brotherly bot" - Justin Yeo Jun Chen whose favourite colour is `#4683b7`

> [!NOTE]
> JustBot is the best 

**JustBot** takes the weight off your shoulders by keeping track of your tasks. It’s literally your bro, helping you navigate your day-to-day responsibilities!

JustBot is: 
- text-based (u gotta bro language ***him***)
- easy to learn
- ~~FAST~~ **SUPER FAST** to use

All you need to do is:

1. [Download it from here](https://www.example.com).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉.

Most importantly, it **FREE**! 

## Features: 
- [x] Managing Tasks 
- [ ] Managing Deadlines (COMING SOON)
- [ ] Reminders (COMING SOON)

## Setup 

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java 
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Justin-Yeo committed:
Add Justbot.java

Justin-Yeo committed:
Add Echo feature for Justbot

Justin-Yeo committed:
Add list feature to Justbot

Justin-Yeo committed:
Refactor commands to Commands.java

Justin-Yeo committed:
Add Task.java and feature to mark/unmark tasks

Justin-Yeo committed:
Add Todo, Event, Deadline and abstract Task

Justin-Yeo committed:
Add automated testing

Justin-Yeo committed:
Add exception handling for list, mark, unmark, todo, deadline and event commands

Justin-Yeo committed:
Add delete command

Justin-Yeo committed:
Add Enums to Commands

Justin-Yeo committed:
Fixed bug of mark/unmark when task has already been mark/unmark

Justin-Yeo committed:
Add save feature

Justin-Yeo committed:
Add Dates and Times feature

Justin-Yeo committed:
Add Parser for Mark and Unmark

Justin-Yeo committed:
Add Parser for Deadline

Justin-Yeo committed:
Refactor EVENT, TODO and DELETE

Justin-Yeo committed:
Refactor print messages into Ui class

Justin-Yeo committed:
Rename TaskFileHandler class to Storage

Justin-Yeo committed:
Add Command class and child classes (Deadline, Event, Delete, List, Mark, Unmark, Todo)

- Implement abstract `Command` class to handle user commands
- Create child classes: `Deadline`, `Event`, `Delete`, `List`, `Mark`, `Unmark`, `Todo`
- Each class encapsulates specific functionality for corresponding command types

Justin-Yeo committed:
Refactor Justbot

Justin-Yeo committed:
Refactor Justbot

Justin-Yeo committed:
Merge branch 'branch-A-MoreOOP'

* branch-A-MoreOOP:
  Refactor Justbot
  Refactor Justbot
  Add Command class and child classes (Deadline, Event, Delete, List, Mark, Unmark, Todo)
  Rename TaskFileHandler class to Storage
  Refactor print messages into Ui class
  Refactor EVENT, TODO and DELETE
  Add Parser for Deadline

Justin-Yeo committed:
Refactor: Organize project files into packages

- Moved Java files into corresponding package directories under src/main/java
- Added package declarations to each Java file based on its directory
- Updated import statements throughout the project to reflect the new package structure

Justin-Yeo committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

Justin-Yeo committed:
Set up Gradle build system for Justbot

Justin-Yeo committed:
Add JUnit tests for Parser class

- Implemented unit tests to cover various command parsing scenarios.
- Tested valid and invalid inputs for todo, deadline, event, and other commands.
- Ensured correct exceptions are thrown for invalid inputs.

Justin-Yeo committed:
Add JUnit tests for Storage class

* Implemented tests for file creation, task saving, and task loading.
* Verified correct handling of valid tasks, malformed lines, and invalid task types.
* Ensured that tasks with malformed dates are skipped during loading.
* Used temporary directory for testing to avoid side effects on the actual file system.

Justin-Yeo committed:
Update build.gradle file

Justin-Yeo committed:
Add Javadoc comments to all classes

Justin-Yeo committed:
Reformat code to adhere to Java coding standards

Justin-Yeo committed:
Add Find command

Justin-Yeo committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add Javadoc comments to all classes

Justin-Yeo committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Reformat code to adhere to Java coding standards

# Conflicts:
#	src/main/java/justbot/Justbot.java
#	src/main/java/justbot/command/Command.java
#	src/main/java/justbot/command/DeadlineCommand.java
#	src/main/java/justbot/command/DeleteCommand.java
#	src/main/java/justbot/command/EventCommand.java
#	src/main/java/justbot/command/MarkCommand.java
#	src/main/java/justbot/command/TodoCommand.java
#	src/main/java/justbot/command/UnknownCommand.java
#	src/main/java/justbot/command/UnmarkCommand.java
#	src/main/java/justbot/task/Deadline.java
#	src/main/java/justbot/task/Task.java
#	src/main/java/justbot/ui/Ui.java

Justin-Yeo committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add Find command

# Conflicts:
#	src/main/java/justbot/Justbot.java
#	src/main/java/justbot/parser/Parser.java
#	src/main/java/justbot/ui/Ui.java

Justin-Yeo committed:
Add relevant checkstyle files

Justin-Yeo committed:
Fix Checkstyle issues

* Adjusted import order according to Checkstyle rules.
* Renamed variables/methods to follow naming conventions.
* Added missing Javadoc comments.
* Removed trailing whitespace and corrected line length.

Justin-Yeo committed:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  Fix Checkstyle issues

clarud commented on src/main/java/justbot/Justbot.java:
```diff
@@ -0,0 +1,64 @@
+     * @param args Command-line arguments passed during the start of the application.
+     */
+    public static void main(String[] args) {
+        new Justbot("/Users/justinyeo/Desktop/CS2103T-ip/data/justbottask.txt").run();
```
Maybe start() / execute() would be more descriptive for initiation?

clarud commented on src/main/java/justbot/command/CommandType.java:
```diff
@@ -0,0 +1,54 @@
+     * @param commandString The string representation of the command type.
+     * @return The CommandType corresponding to the given string, or UNKNOWN if no match is found.
+     */
+    public static CommandType fromString(String commandString) {
```
I think getCommandType might be more apt in this case

SwaminathanViswa replied:
I also feel that getCommandType() or getCommand() would be more suitable and make it easier to understand

clarud commented on src/main/java/justbot/command/DeleteCommand.java:
```diff
@@ -0,0 +1,57 @@
+     * @param deleteNumber The 1-based index of the task to be deleted from the task list.
+     */
+    public DeleteCommand(int deleteNumber) {
+        this.deleteNumber = deleteNumber;
```
Can deleteNumber be more descriptive such as taskIndexToDelete.

clarud commented on src/main/java/justbot/command/MarkCommand.java:
```diff
@@ -0,0 +1,58 @@
+     * @param markNumber The 1-based index of the task to be marked as done.
+     */
+    public MarkCommand(int markNumber) {
+        this.markNumber = markNumber;
```
Similar to the other one, I think markNumber can be more descriptive such as taskIndexToMark

clarud reviewed (commented):
Code Naming is very well done, difficult to find issues.

SwaminathanViswa commented on src/main/java/justbot/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,58 @@
+     *
+     * @param unmarkNumber The 1-based index of the task to be unmarked as not done.
+     */
+    public UnmarkCommand(int unmarkNumber) {
```
As with the other comments mentioned above, I feel that taskNumberToUnmark is more appropriate as it is more descriptive

SwaminathanViswa commented on src/main/java/justbot/task/Task.java:
```diff
@@ -0,0 +1,85 @@
+     *
+     * @return {@code true} if the task is done, {@code false} otherwise.
+     */
+    public boolean getIsDone() {
```
do you think if isDone() is better? 🤔

SwaminathanViswa commented on src/main/java/justbot/command/MarkCommand.java:
```diff
@@ -0,0 +1,58 @@
+     *
+     * @param markNumber The 1-based index of the task to be marked as done.
+     */
+    public MarkCommand(int markNumber) {
```
Similar to the comment below, I also believe that taskIndexToMark will be more descriptive! 😄

SwaminathanViswa commented on src/main/java/justbot/storage/Storage.java:
```diff
@@ -0,0 +1,177 @@
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
```
Maybe before this line, you could give a line break so that the ordering is more obvious!

SwaminathanViswa reviewed (commented):
Very well done in terms of coding standards and there is hardly any coding errors to catch! Well done! 💯 😄

Justin-Yeo committed:
Add JavaFX resources and refactor for GUI

OrangeCatLoves commented:
Nice classification of various commands into different classes. Adhered to OOP principles!

Justin-Yeo committed:
Fix out-of-bound issue for Delete Command
* Added validation to ensure the task index is within bounds before deletion.
* Prevented deletion from being attempted on an empty task list.
* Improved error messages for invalid task numbers.

Justin-Yeo committed:
Copy code
Update dialog box background and text colors for improved UI design

* Change bot dialog box color to white.
*Adjusted text color for better contrast and readability.

Justin-Yeo committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Copy code Update dialog box background and text colors for improved UI design
  Fix out-of-bound issue for Delete Command * Added validation to ensure the task index is within bounds before deletion. * Prevented deletion from being attempted on an empty task list. * Improved error messages for invalid task numbers.
  Add JavaFX resources and refactor for GUI

Justin-Yeo committed:
Add VarArgs to include multiple keywords for find and GUI bug fix

Justin-Yeo committed:
Merge branch 'branch-A-Varargs'

* branch-A-Varargs:
  Add VarArgs to include multiple keywords for find and GUI bug fix

Justin-Yeo committed:
Add assertions
*Included assertions for key classes such as Parser class and Main class

Justin-Yeo committed:
Extract task matching function into TaskMatcher class

Justin-Yeo committed:
Add HashMap for CommandType class
*Improves readability of the fromString method in the CommandType class

Justin-Yeo committed:
Refactor TaskMatcher to use Java Streams for improved readability and performance

* Replaced for-loops with Java Streams in findMatchingTasks and doesTaskMatchAnyKeyword methods.
* Improved code readability with the use of JavaStreams

Justin-Yeo committed:
Merge pull request #1 from Justin-Yeo/branch-A-Assertions

Add assertions

Justin-Yeo committed:
Merge pull request #2 from Justin-Yeo/branch-A-CodeQuality

Branch a code quality

Justin-Yeo committed:
Merge pull request #3 from Justin-Yeo/branch-A-Streams

Refactor TaskMatcher to use Java Streams for improved readability and…

Justin-Yeo committed:
Merge branch 'master' of https://github.com/Justin-Yeo/ip

* 'master' of https://github.com/Justin-Yeo/ip:
  Refactor TaskMatcher to use Java Streams for improved readability and performance
  Add HashMap for CommandType class *Improves readability of the fromString method in the CommandType class
  Extract task matching function into TaskMatcher class
  Add assertions *Included assertions for key classes such as Parser class and Main class

Justin-Yeo committed:
Update gitignore
*Added data file to gitignore

Justin-Yeo committed:
Add chronological sorting for tasks
* Implemented a sort function which is called after any addition/deletion of tasks to ensure tasklist is sorted in chronological order

Justin-Yeo committed:
Merge pull request #4 from Justin-Yeo/branch-C-Sort

Add chronological sorting for tasks

Justin-Yeo committed:
Parser: Refactor parseCommand method for better readability

The parseCommand method in the Parser class is currently too long and
contains all the command logic within a single method.

Long methods reduce code readability and make maintenance more
difficult. Breaking down this method into smaller, more focused methods
improves clarity and makes the code easier to understand and extend.

Justin-Yeo committed:
Parser: Refactor Parser class for more SLAP

Applied SLAP more aggressively

Justin-Yeo committed:
Add Javadocs for Parser class

Justin-Yeo committed:
Update ParserTest

Justin-Yeo committed:
Storage: Refactor Storage class

Applied SLAP for loadTasks method

Justin-Yeo committed:
Add Javadoc for sortTaskChronologically method in Tasklist

Justin-Yeo committed:
Update README

Justin-Yeo committed:
Add images for README

Justin-Yeo committed:
Merge pull request #6 from Justin-Yeo/branch-A-UserGuide

Branch a user guide

Justin-Yeo committed:
Update README example usage image

Justin-Yeo committed:
Add JavaDocs to Parser class

Justin-Yeo committed:
Refactor Storage class

Justin-Yeo committed:
Add check for missing data file
